File: test-suite/src/test/java/io/micronaut/security/docs/securedexpressions/ExampleController.java
package io.micronaut.security.docs.securedexpressions;

import io.micronaut.context.annotation.Requires;
import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Produces;
import io.micronaut.security.annotation.Secured;
import java.security.Principal;

@Requires(property = "spec.name", value = "docexpressions")
//tag::exampleControllerExpressions[]
@Controller("/authenticated")
public class ExampleController {

    @Secured("#{ user?.attributes?.get('email') == 'sherlock@micronaut.example' }")
    @Produces(MediaType.TEXT_PLAIN)
    @Get("/email")
    public String authenticationByEmail(Principal principal) {
        return principal.getName() + " is authenticated";
    }
}
//end::exampleControllerExpressions[]

File: test-suite-python/src/test/python/micronaut/security/docs/securedexpressions/ExampleController.py
from java.security import Principal
from micronaut.context.annotation import Requires
from micronaut.http import MediaType
from micronaut.http.annotation import Controller, Get, Produces
from micronaut.security.annotation import Secured


@Requires(property="spec.name", value="docexpressions")
# tag::exampleControllerExpressions[]
@Controller("/authenticated")
class ExampleController:

    @Secured("#{ user?.attributes?.get('email') == 'sherlock@micronaut.example' }")
    @Produces(MediaType.TEXT_PLAIN)
    @Get("/email")
    def authentication_by_email(self, principal: Principal) -> str:
        return principal.getName() + " is authenticated"
# end::exampleControllerExpressions[]
